docs/docs/adapters/kysely.md

```diff
@@ -0,0 +1,275 @@
+npm install --save-dev @types/pg
+```
+
+Configure your NextAuth.js to use the Kysely Adapter:
+
+```javascript title="pages/api/auth/[...nextauth].js"
+import NextAuth from "next-auth"
+import GoogleProvider from "next-auth/providers/google"
+import { KyselyAdapter } from "@next-auth/kysely-adapter"
+import { db } from "../../../db"
+
+export default NextAuth({
+  adapter: KyselyAdapter(db),
+  providers: [
+    GoogleProvider({
+      clientId: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+    }),
+  ],
+})
+```
+
+## Setup
+
+[Kysely](https://github.com/koskimas/kysely) (pronounce “Key-Seh-Lee”) is a type-safe and autocompletion-friendly TypeScript SQL query builder inspired by Knex. To use Kysely, you define interfaces for each of your database tables and pass them to the `Kysely` constructor.
+
+### Create type definitions and the Kysely instance
+
+This Adapter exports a wrapper around the original Kysely class, `AuthedKysely`, that can be used to provide an additional level of type safety. While using it isn't required, it is recommended as it will verify that the database interface has all the fields that NextAuth.js requires.
+
+:::note
+An alternative to manually defining types is generating them from the database schema using [kysely-codegen](https://github.com/RobinBlomberg/kysely-codegen). When using the generated types with AuthedKysely, import `Codegen` and pass it as the second generic arg:
+
+```
+import type { Codegen } from "@next-auth/kysely-adapter"
+new AuthedKysely<Database, Codegen>(...)
+
+```
+
+:::
+
+```ts title="db/index.ts"
+import { PostgresDialect } from "kysely"
+import type { Generated } from "kysely"
+import { Pool } from "pg"
+import { AuthedKysely } from "@next-auth/kysely-adapter"
+
+interface User {
+  id: Generated<string>
+  name: string | null
+  email: string | null
+  emailVerified: Date | null
+  image: string | null
+}
+
+interface Account {
+  id: Generated<string>
+  userId: string
+  type: string
+  provider: string
+  providerAccountId: string
+  refresh_token: string | null
+  access_token: string | null
+  expires_at: number | null
+  token_type: string | null
+  scope: string | null
+  id_token: string | null
+  session_state: string | null
+  oauth_token_secret: string | null
+  oauth_token: string | null
+}
+
+interface Session {
+  id: Generated<string>
+  userId: string
+  sessionToken: string
+  expires: Date
+}
+
+interface VerificationToken {
+  identifier: string
+  token: string
+  expires: Date
+}
+
+export interface Database {
+  User: User
+  Account: Account
+  Session: Session
+  VerificationToken: VerificationToken
+}
+
+export const db = new AuthedKysely<Database>({
+  dialect: new PostgresDialect({
+    pool: new Pool({
+      host: process.env.DATABASE_HOST,
+      database: process.env.DATABASE_NAME,
+      user: process.env.DATABASE_USER,
+      password: process.env.DATABASE_PASSWORD,
+    }),
+  }),
+})
+```
+
+### Create migrations
```

Comment:
is this next-auth dependent? seems like a thing that you expect to find in kysely's docs and not here?

Reply:
I think the example schema might be helpful as a starting point — I've updated it to be pretty close to the other adapters that have had their docs updated.